### PR TITLE
first steps script on scrapeme.live

### PR DESCRIPTION
**Pull request by shuhaodo**
Script for the "first steps" tutorial using the latest SDK API

https://agentql-docs.vercel.app/getting-started/first-steps

### CODE REVIEW

**Review by rachelnabors — APPROVED** [review]

**Comment by rachelnabors** [comment]
@shuhaodo Should we link to this repo on the site as well, for folks who want to "fork and play?"

**Review by colriot — CHANGES_REQUESTED** [review]
Added comments for discussion

File: examples/first_steps/first_steps.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        price_currency
+        products[] {
+            name
+            price
+        }
+    }"""
+
+    # Extract data using AgentQL Data Query API
+    # API Doc: TODO add API Doc link
```
[the remark this answers is not in the document]
Comment (shuhaodo) [reply examples/first_steps/first_steps.py:65]: good point, removed

File: examples/first_steps/first_steps.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# This enables the use of the AgentQL Smart Locator and Data Query API
+from agentql.ext.playwright.sync_api import Page
+
+URL = "https://scrapeme.live/shop"
+
+
+def main():
+    """Main function."""
+    with sync_playwright() as playwright:
+        browser = playwright.chromium.launch(headless=False)
```
[the remark this answers is not in the document]
Comment (shuhaodo) [reply examples/first_steps/first_steps.py:17]: yes!

File: examples/first_steps/first_steps.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        page: Page = browser.new_page()
```
[the remark this answers is not in the document]
Comment (shuhaodo) [reply examples/first_steps/first_steps.py:18]: No linter error for me. What's the error on your side?
image

**Comment by frankfeng98** [comment]
@rachelnabors @shuhaodo I took the liberty of updating this PR since it will be downloaded with `agentql init` command. The main changes are:

(1) All the queries and natural language prompt are created as constant variables, like other examples.
(2) `query_elements()` API is added into the example. So now, first steps examples have all three APIs (elements query, data query, Smart Locator)

Let me know your thoughts! Once this is done, I will also create an async version of the first step script to showcase our asynchronous library.

File: examples/first_steps/first_steps.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    """Extract product data.
+
+    Args:
+        page (Page): The Playwright page object to interact with the browser.
+        search_key_word (str): The product to search for.
+
+    Returns:
+        dict: The product data extracted from the page.
+    """
+    # Find DOM element using AgentQL Elements Query API
```
Comment (rachelnabors) [inline examples/first_steps/first_steps.py:74]:
```suggestion
    # Find DOM element using AgentQL Elements Query API query_elements()
    # query_elements() returns matching HTML elements
```

File: examples/first_steps/first_steps.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    response = page.query_elements(SEARCH_BOX_QUERY)
+
+    # Interact with the element using Playwright API
+    # API Doc: https://playwright.dev/python/docs/input#text-input
+    response.search_product_box.type(search_key_word, delay=200)
+    page.keyboard.press("Enter")
+
+    # Extract data using AgentQL Data Query API
```
Comment (rachelnabors) [inline examples/first_steps/first_steps.py:82]:
```suggestion
    # Extract data using AgentQL Data Query API query_data()
    # query_date() returns data from matching elements, not the elements themselves
```

**Review by rachelnabors — DISMISSED** [review]
Looks good! Wanted to expand on each of these three selector options. I think it's worth clarifying them further in the docs

**Review by frankfeng98 — APPROVED** [review]

**Pull request merged**